pkg/controllers/metrics/node/controller.go

```diff
@@ -157,6 +191,43 @@ func (c *Controller) Register(_ context.Context, m manager.Manager) error {
 		Complete(singleton.AsReconciler(c))
 }
 
+func buildClusterUtilizationMetric(nodes state.StateNodes) []*metrics.StoreMetric {
+
+	// Aggregate resources allocated/utilized for all the nodes and pods inside the nodes
+	allocatableAggregate, utilizedAggregate := corev1.ResourceList{}, corev1.ResourceList{}
+
+	for _, node := range nodes {
+		resources.MergeInto(allocatableAggregate, node.Allocatable())
+		resources.MergeInto(utilizedAggregate, node.PodRequests())
+	}
+
+	res := make([]*metrics.StoreMetric, 0, len(allocatableAggregate))
+
+	for resourceName, allocatableResource := range allocatableAggregate {
+
+		if allocatableResource.Value() == 0 {
+			// This zero check may be unnecessary. I'm erring towards caution.
```

Comment:
I'm not sure this 0 check is necessary because the resources I have in mind (cpu, memory, storage) should never be 0.

As for pod resources, I'm not sure if the default value is 0 (will be problematic) or just not existing (what I expect and shouldn't be problematic).

Finally, I'm not sure if there are other resources I don't know.

Reply:
I'd agree that if this value is 0 that we shouldn't publish it